src/main/kotlin/device/application/CreateDevice.kt

```diff
@@ -0,0 +1,32 @@
+package device.application
+
+import device.domain.Device
+import device.domain.DeviceIdentifier
+import device.domain.DeviceName
+import device.domain.DeviceRepository
+import device.domain.exceptions.AlreadyExistingDevice
+import device.domain.exceptions.InvalidUUIDException
+
+class CreateDevice(private val repository: DeviceRepository) {
+
+    /***
+     * Creates a new device from primitives and saves it to the DeviceRepository
+     * @throws InvalidUUIDException if the provided id is not a valid UUIDv4 identifier
+     * @throws AlreadyExistingDevice if a new device already exists
```

Comment:
The documentation misses that the method throws IllegalDeviceNameException if the device name isn't correct.